docs/howtos/converting-a-component-to-uniffi.md

```diff
@@ -0,0 +1,287 @@
+
+Find the places in the Rust code that are using these message types to return structured data.
+In simple cases, you may be able to directly replace uses of `msg_types::ExampleStruct` with
+the corresponding `crate::ExampleStruct` from your public API.
+For more complex cases, you may find it helpful to define an `Into` mapping between the
+UniFFI dictionary/enum in the crate's public interface, and a more complex struct designed
+for internal use.
+
+Once you have replaced all uses of the `msg_types` structs in the Rust code:
+
+* Delete `./src/<component_name>_msg_types.proto`.
+* Delete `./src/mozilla.appservices.<component_name>.protobuf.rs`, which is generated from the `.proto` file.
+* Remote `prost` and `prost-derive` from the crate's dependencies.
+* Delete the crate from the list in `/tools/protobuf_files.toml`.
+
+If it happen to find that you've deleted the last crate from the list in `protobuf_files.toml`,
+congratulations! You've successfully removed protocol buffers from this repo entirely, and should
+file a bug to track the complete removal of protobuf from our tooling and dependency chain.
+
+## Document the Public API in the Rust code
+
+Write consumer-facing documentation on the public API in `lib.rs` using Rust's standard
+[rustdoc](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html) conventions
+and tools. The `fxa-client` crate may serve as a good example.
+
+You can view the generated documentation by running:
+
+```
+cargo doc --no-deps --open
+```
+
+In future, we intend to automatically extract documentation from the Rust code
+and make it easily available to consumers of the generated bindings.
+
+## Set up the Kotlin wrapper
+
+It's easiest to start by removing all of the hand-written Kotlin code under `android/src/main/java`
+and then restoring parts of it later if necessary. Leave the `AndroidManifest.xml` file and any tests
+in place.
+
+Delete the `android/build.gradle` file and then follow the instructions for [adding Kotlin bindings
+for a new component](adding-a-new-component.md#the-kotlin-bindings) to create a new `build.gradle`
+file and a corresponding `uniffi.toml`.
+
+This should be all that's required to set up UniFFI to build the Kotlin bindings. Try building
+the Android package to confirm:
+
+* `./gradlew <component_name>:assembleDebug`
+
+The UniFFI-generated Kotlin code will be under `./android/build/generated/source/uniffi/` and
+may be useful for debugging.
+
+If there are existing Kotlin tests for the component, the next step is to get those passing:
+
+* `./gradlew <component_name>:test`
+
+It is normal and expected for the autogenerated bindings to be subtly different from the
+previous hand-written ones. For example, UniFFI insists on CAMEL_CASING variant names in
+Kotlin enums while the hand-written code does not to this consistently. Some components
+also have small naming differences between the Rust code and the hand-written Kotlin bindings,
+which UniFFI will not allow.
```

Comment:
If you take the previous "don't be afraid" suggestions, maybe add here "As noted above"?